src/utils.js

```diff
@@ -209,3 +216,100 @@ export function createModelledFunctionCall(
     }
   };
 }
+
+export function isBindingMutationOutsideFunction(
+  binding: Binding,
+  bindingEntry: BindingEntry,
+  effects: Effects,
+  F: FunctionValue
+): boolean {
+  // If either the previous values are uninitialized then this is not a mutation
+  if (binding.value === undefined || bindingEntry.value === undefined) {
+    return false;
+  }
+  // If the binding has a "strict" property then it's const and can never mutate
+  if (binding.strict !== undefined) {
+    return false;
+  }
+  let env = F.$Environment;
+  let bindingName = binding.name;
+  let bindingEnv = binding.environment;
+  // If the bindingEnv is the same env, we know the mutation was only local
+  if (bindingEnv instanceof FunctionEnvironmentRecord && bindingEnv.$FunctionObject === F) {
+    return false;
+  }
+  // Check if the binding was mutated in a scope of F or its parents
```

Comment:
Not "F or its parents" "F's parents", we return false if binding was mutated in F's scope above.